zerver/views/auth.py

```diff
@@ -252,9 +253,12 @@ def finish_desktop_flow(request: HttpRequest, user_profile: UserProfile,
     data = {'email': user_profile.delivery_email,
             'subdomain': realm.subdomain}
     token = store_login_data(data)
-
-    return create_response_for_otp_flow(token, otp, user_profile,
-                                        encrypted_key_field_name='otp_encrypted_login_key')
+    response = create_response_for_otp_flow(token, otp, user_profile,
+                                            encrypted_key_field_name='otp_encrypted_login_key')
+    context = {'desktop_url': response['Location'],
+               'browser_url': user_profile.realm.uri + '/accounts/login/subdomain/' + token,
+               'realm_icon_url': realm_icon_url(realm)}
+    return render(request, 'zerver/desktop_redirect.html', context=context)
```

Comment:
I thought this would end up having to be more messy, I like how simple this change came out :slightly_smiling_face: